backend/core/src/app.module.ts

```diff
@@ -46,7 +46,7 @@ export function applicationSetup(app: INestApplication) {
       credentials: true,
       origin: false,
     }
-    if (allowList.indexOf(req.header("Origin")) !== -1) {
+    if (allowList.indexOf(req.header("Origin")) !== -1 || process.env.NODE_ENV !== "production") {
```

Comment:
There's space to do something more sophisticated here, I somewhat worry this will remove test coverage from our dev environments of this code path, but curious for perspectives on whether or not that is a blocking issue / vibes on if that feels like an issue at all

Reply:
I think I'm a little hesitant to blanket turn off cors checks on non prod environments

After looking at the preview links this pr generated:
https://deploy-preview-3564--bloom-demo-public.netlify.app/
https://deploy-preview-3564--bloom-exygy-dev.netlify.app/

I think I'm more in favor of changing this from a `.indexOf()` check to a regex that checks to see from where the request is originating from.

I think knowing that the hosted urls are:
https://partners.bloom.exygy.dev/ and https://bloom.exygy.dev/

I'm wondering if we could construct a regex that covers all 4 possibilities (we also need to cover the proxy and the backend url)